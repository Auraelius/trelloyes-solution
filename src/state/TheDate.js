import React from 'react';

class TheDate extends React.Component {
  constructor(props) {
    super(props); // required for all react components
    this.state = { datetime: new Date() };
    // console.log('TheDate: constructor')
  }

  // called once when the componet is first added to the DOM, or "mounted"
  componentDidMount() {
    // use setInterval() run this callback function every second
    setInterval(() => { 
      // you must only update the state object with setState() method 
      // so that React can, uh, react to the new state
      this.setState({
        datetime:new Date() // what happens to all the old Date objs? garbage collection?
      });
      // console.log('TheDate: setInterval callback')
    }, 1000);
    // console.log('TheDate: componentDidMount')
  }
  // called when the component is about to be removed from the DOM
  componentWillUnmount() {
    clearInterval(this.interval);
    // console.log('TheDate: componentWillUnmount')
    
  }

  render() {
    // console.log('TheDate: render')
    // console.log(this.state);
    return (
      <div>
        {this.state.datetime.toLocaleString()}
      </div>
    );
  }
}

export default TheDate