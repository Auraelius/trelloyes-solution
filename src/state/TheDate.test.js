import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import TheDate from './TheDate'

describe(`TheDate Component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TheDate />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  // we know this test doesn't work due to the constantly-updating date, but we haven't been told how to fix it yet
  it.skip('renders the UI as expected', () => {
    expect(
      renderer.create(<TheDate />).toJSON()
    ).toMatchSnapshot()
  })
})