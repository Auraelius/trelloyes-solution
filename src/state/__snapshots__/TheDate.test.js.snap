// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheDate Component renders the UI as expected 1`] = `
<div>
  5/11/2019, 6:23:50 PM
</div>
`;
